### ISSUE TITLE
cheat.dat support other language

### ISSUE THREAD

**Issue by king2015** [libretro/mame2003-plus-libretro#1179]
I added a Chinese description in cheat.dat, but the display is garbled.
I tried utf-8 and gb2312 with the same result,how to support,How to support this.

**Comment by mahoneyt944** [comment]
I haven't personally reviewed this area of the code. But I would assume the cheat engine is using a list of known characters somewhere. You'd probably have to add the Chinese characters to it's internal structures to support.

Here's an array I found by scanning through the code, not sure if this is related directly or not but as you can see it's predetermined with a set list of keys. I'm guessing somewhere along the lines it's encounter the Chinese characters as unknown?
https://github.com/libretro/mame2003-plus-libretro/blob/8f69baf45f8e42873f10eefd749763ec9482e3b5/src/cheat.c#L2841-L2865

**Comment by mahoneyt944** [comment]
So I believe the font is baked into the core with no real way to do this without adding an entirely new font. I don't see an easy way to do this. @grant2258  @arcadez2003 either of you see any viable solutions for this?

**Comment by markwkidd** [comment]
There is a RetroArch feature that sends screenshots to an AI translation service or software. That's only useful in some circumstances but might be a help to some people who don't read English trying to use cheats: https://docs.libretro.com/guides/ai-service/

It's a workaround a best.

**Comment by mahoneyt944** [comment]
Yeah, otherwise I think we'd have to include the new font in the core then run cheat.dat through a translator. Retroarch supports many languages but idk if we'd be able to piggyback on one libretro has or just include the raw font in the core.
